main/handlersettings_test.go

```diff
@@ -194,3 +200,48 @@ func Test_toJSONUmarshallForManagedIdentity(t *testing.T) {
 	h = handlerSettings{publicSettings{}, *protSettings}
 	require.Error(t, h.validate(), "settings should be invalid")
 }
+
+func Test_protectedSettingsTest(t *testing.T) {
+	//set up test direcotry + test files
+	testFolderPath := "/config"
+	settingsExtensionName := ".settings"
+
+	err := createTestFiles(testFolderPath, settingsExtensionName)
+	assert.NoError(t, err)
+
+	err = cleanUpSettings(testFolderPath)
+	assert.NoError(t, err)
+
+	fileName := ""
+	for i := 0; i < 3; i++ {
+		fileName = filepath.Join(testFolderPath, strconv.FormatInt(int64(i), 10)+settingsExtensionName)
+		content, err := ioutil.ReadFile(fileName)
+		assert.NoError(t, err)
+		assert.Equal(t, len(content), 0)
+	}
+
+	// cleanup
+	defer os.RemoveAll(testFolderPath)
+}
+
+func createTestFiles(folderPath, settingsExtensionName string) error {
```

Comment:
should add a test where we have one of the files open so we can't modify it (extension platform should have a similar test whose logic can be copied)